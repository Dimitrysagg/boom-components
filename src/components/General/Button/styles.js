import styled from 'styled-components'

export const DefaultBtn = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 35px;
  padding: 0 20px;
  border: 0;
  border-radius: 2px;
  font-size: 16px;
  cursor: pointer;
  transition: opacity 150ms ease-in-out;

  &:hover {
    opacity: 0.8;
  }

  &:disabled {
    cursor: no-drop;

    &:hover {
      opacity: 1;
    }
  }
`

export const Icon = styled.span`
  svg {
    min-height: 10px;
    height: 100%;
  }
`

export const Text = styled.span`
  flex: 1;
`
