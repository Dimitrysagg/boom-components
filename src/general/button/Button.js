import React from "react";
import PropTypes from "prop-types";
import Loader from "../../shared/loader";

import {
  DefaultBtn,
  Icon,
  LoadingFeedback,
  ButtonContent,
  ButtonChildren
} from "./styles";

import { noop } from "../../helpers";

/**
 * Component to trigger an operation.
 *
 * @param {*} { children, textStyle, icon, iconStyle, htmlType, type, ...props }
 * @returns {Component}
 */
const Button = ({
  children,
  textStyle,
  icon,
  iconStyle,
  htmlType,
  type,
  loading,
  loadingColor,
  loadingSize,
  disabled,
  className,
  ...props
}) => (
  <DefaultBtn
    type={htmlType}
    colorType={type}
    disabled={loading || disabled}
    loadingSize={loadingSize}
    className={`${className} ${loading ? "loading" : ""}`}
    {...props}
  >
    <ButtonContent>
      {loading && (
        <LoadingFeedback>
          <span>
            <Loader loadingSize={loadingSize} loadingColor={loadingColor} />
          </span>
        </LoadingFeedback>
      )}
      <ButtonChildren style={textStyle}>
        {icon && <Icon style={iconStyle}>{icon}</Icon>}
        {children}
      </ButtonChildren>
    </ButtonContent>
  </DefaultBtn>
);

Button.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  /** set the class of button */
  className: PropTypes.string,
  /** disabled state of the button */
  disabled: PropTypes.bool,
  /** redirect url of link button */
  href: PropTypes.string,
  /** HTML type of button */
  htmlType: PropTypes.oneOf(["button", "reset", "submit"]),
  /** set the icon of button */
  icon: PropTypes.string,
  /** icon style */
  iconStyle: PropTypes.object,
  /** set the loading status of button */
  loading: PropTypes.bool,
  /** loading icon size */
  loadingSize: PropTypes.number,
  /** loading icon color */
  loadingColor: PropTypes.string,
  /** set the handler of click event */
  onClick: PropTypes.func,
  /** make background transparent and invert text and border colors */
  outlined: PropTypes.bool,
  /** make button circular */
  rounded: PropTypes.bool,
  /** size of the button */
  size: PropTypes.oneOf(["small", "default", "large"]),
  /** fit button width to its parent width */
  stretched: PropTypes.bool,
  /** set the style of button */
  style: PropTypes.object,
  /** same as target attr of a but works only when href is specified */
  target: PropTypes.string,
  /** set the pattern of button */
  type: PropTypes.oneOf(["main", "primary"])
};

Button.defaultProps = {
  className: null,
  disabled: false,
  htmlType: "button",
  href: null,
  icon: null,
  iconStyle: {},
  loading: false,
  loadingColor: "#fff",
  loadingSize: 18,
  outlined: false,
  onClick: noop,
  rounded: false,
  size: "default",
  stretched: false,
  style: {},
  target: null,
  type: "main"
};

export default Button;
