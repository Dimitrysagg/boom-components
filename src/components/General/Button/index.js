import React from 'react'
import PropTypes from 'prop-types'

import { DefaultBtn, Icon, Text } from './styles'

const Button = ({ className, children, textStyle, style, type, icon, iconStyle, disabled, onClick }) => {
  return (
    <DefaultBtn type={type} className={className} onClick={onClick} disabled={disabled} style={style}>
      {icon && <Icon style={iconStyle}>{icon}</Icon>}
      <Text style={textStyle}>{children}</Text>
    </DefaultBtn>
  )
}

Button.propTypes = {
  className: PropTypes.string,
  children: PropTypes.string.isRequired,
  textStyle: PropTypes.object,
  style: PropTypes.object,
  type: PropTypes.string,
  icon: PropTypes.element,
  iconStyle: PropTypes.object,
  disabled: PropTypes.bool,
  onClick: PropTypes.func
}

export default Button
