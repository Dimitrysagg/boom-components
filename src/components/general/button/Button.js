import React from "react";
import PropTypes from "prop-types";

import { DefaultBtn, Icon } from "./styles";
import { noop } from "../../../helpers";

/**
 * Component to trigger an operation.
 *
 * @param {*} { children, textStyle, icon, iconStyle, htmlType, type, ...props }
 * @returns {Component}
 */
const Button = ({ children, textStyle, icon, htmlType, type, ...props }) => (
  <DefaultBtn type={htmlType} colorType={type} {...props}>
    {icon && <Icon>{icon}</Icon>}
    {children}
  </DefaultBtn>
);

Button.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  /** set the class of button */
  className: PropTypes.string,
  /** disabled state of the button */
  disabled: PropTypes.bool,
  /** redirect url of link button */
  href: PropTypes.string,
  /** HTML type of button */
  htmlType: PropTypes.oneOf(["button", "reset", "submit"]),
  /** set the icon of button */
  icon: PropTypes.string,
  /** set the loading status of button */
  loading: PropTypes.bool,
  /** set the handler of click event */
  onClick: PropTypes.func,
  /** make background transparent and invert text and border colors */
  outlined: PropTypes.bool,
  /** make button circular */
  rounded: PropTypes.bool,
  /** size of the button */
  size: PropTypes.oneOf(["small", "default", "large"]),
  /** fit button width to its parent width */
  stretched: PropTypes.bool,
  /** set the style of button */
  style: PropTypes.object,
  /** same as target attr of a but works only when href is specified */
  target: PropTypes.string,
  /** set the pattern of button */
  type: PropTypes.oneOf(["main", "primary"])
};

Button.defaultProps = {
  className: null,
  disabled: false,
  htmlType: "button",
  href: null,
  icon: null,
  loading: false,
  outlined: false,
  onClick: noop,
  rounded: false,
  size: "default",
  stretched: false,
  style: {},
  target: null,
  type: "main"
};

export default Button;
