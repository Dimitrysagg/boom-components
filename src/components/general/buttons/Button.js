import React from 'react'
import PropTypes from 'prop-types'

import { DefaultBtn, Icon, Text } from './styles'

/**
 * Component to trigger an operation.
 *
 * @param {*} { className, children, textStyle, style, type, icon, iconStyle, disabled, onClick }
 * @returns {Component}
 */
const Button = ({ children, textStyle, icon, iconStyle, ...props }) => (
  <DefaultBtn {...props}>
    {icon && <Icon style={iconStyle}>{icon}</Icon>}
    <Text style={textStyle}>{children}</Text>
  </DefaultBtn>
)

Button.propTypes = {
  /** text of button */
  children: PropTypes.string.isRequired,
  /** set the class of button */
  className: PropTypes.string,
  /** disabled state of the button */
  disabled: PropTypes.bool,
  /** set the icon of button */
  icon: PropTypes.element,
  /** set the style of icon */
  iconStyle: PropTypes.object,
  /** set the handler of click event */
  onClick: PropTypes.func,
  /** set the style of button */
  style: PropTypes.object,
  /** set the style of text */
  textStyle: PropTypes.object,
  /** HTML type of button */
  type: PropTypes.oneOf(['button', 'reset', 'submit'])
}

Button.defaultProps = {
  children: '',
  disabled: false,
  type: 'button'
}

export default Button
