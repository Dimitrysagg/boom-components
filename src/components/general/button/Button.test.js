import React from 'react'
import { mount } from 'enzyme'

import Button from '../buttons'

describe('Button', () => {
  test('Snapshot', () => {
    const wrapper = mount(<Button>Test</Button>)
    const tree = wrapper.find(Button)

    expect(tree).toMatchSnapshot()
  })
})
