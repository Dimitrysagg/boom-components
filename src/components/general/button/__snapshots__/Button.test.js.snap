// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot 1`] = `
<Button
  disabled={false}
  htmlType="button"
  type="default"
>
  <styled.button
    colorType="default"
    disabled={false}
    type="button"
  >
    <StyledComponent
      colorType="default"
      disabled={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "gZAtua",
            "rules": Array [
              "
  display: inline-block;
  align-items: center;
  justify-content: center;
  height: 35px;
  padding: 0 20px;
  border: 0;
  border-radius: 2px;
  font-size: 16px;
  cursor: pointer;
  transition: opacity 150ms ease-in-out;
  background-color: ",
              [Function],
              ";
  color: ",
              [Function],
              ";
  margin-right: 8px;
  margin-bottom: 12px;

  &:hover {
    opacity: 0.8;
  }

  &:disabled {
    cursor: no-drop;

    &:hover {
      opacity: 1;
    }
  }
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      type="button"
    >
      <button
        className="sc-bdVaJa gZAtua"
        disabled={false}
        type="button"
      >
        <styled.span>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "gChKYK",
                  "rules": Array [
                    "
  flex: 1;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-htpNat gChKYK"
            >
              Test
            </span>
          </StyledComponent>
        </styled.span>
      </button>
    </StyledComponent>
  </styled.button>
</Button>
`;
